src/reader/reader.py

```diff
@@ -0,0 +1,16 @@
+class Reader():
+    """
+    Reader is a generic wrapper for reading data from a source that supports a read() method.
+    """
+    def __init__(self, source):
+        self.source = source
+
+    def read(self):
+        """
+        Read data from the source.
+        """
+        try:
+            data = self.source.read()
+        except Exception as e:
+            raise ValueError("Error reading from source: " + str(e))
+        return data
```

Comment:
I like this abstraction as a starting point! One thing we'll have to think about is how to deal with a corpus as it becomes very large -- one thing we might want to be able to do in the future is to have the `Reader` accept as a parameter some kind of "starting position" (e.g. a filename, a subdirectory, maybe even a Kafka topic, etc). This would give us the opportunity to read in and parse only a subset of the corpus, or alternatively, to distribute the reading/parsing effort across multiple threads.

Reply:
Thanks for the suggestion, I think that would be a useful parameter to have and it seems to be a generic thing that all types of stream readers want to do.